### PR TITLE
feature/helm-chart

### PR DESCRIPTION
**Pull request by dmakeroam**
I added Helm chart to help quickly starting with msi-acrpull on Kubernetes with the following changelog

chore - add helm index
docs: removed helm 2 step
chore - Added helm chart for msi-acrpull

### CODE REVIEW

**Comment by dmakeroam** [comment]
@samkreter @yangl900 Could you help to review on this?

**Comment by dmakeroam** [comment]
Any updates on this review? @weng271190436

File: charts/msi-acrpull/crds/crd.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.2.5
+    "helm.sh/hook": crd-install
```
Comment (weng271190436) [inline charts/msi-acrpull/crds/crd.yaml:6]: is this supposed to work with helm2?

Comment (weng271190436) [reply charts/msi-acrpull/crds/crd.yaml:6]: in the README you said `helm install my-release msi-acrpull/msi-acrpull` I think this my-release argument only works for helm3 but helm.sh/hook: crd-install only works for helm2?

File: README.md [hunk 2]
```diff
@@ -9,10 +9,20 @@ Run following command to install latest build from main branch. It will install
...
+helm install msi-acrpull msi-acrpull/msi-acrpull -f ./values.yaml --version v0.1.0-alpha -n msi-acrpull-system
+```
+
 # How to use
 > NOTE: following steps assumes you already have:
 > 1) An Kubernetes cluster, and have user assigned managed identities on node pool VMSS.
-> 1) An ACR, and the user assigned identity has [AcrPull](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-roles#pull-image) role assigned on ACR.
+> 2) An ACR, and the user assigned identity has [AcrPull](https://docs.microsoft.com/en-us/azure/container-registry/
+container-registry-roles#pull-image) role assigned on ACR.
+> 3) The user assigned identity has been mapped to msi-acrpull pod with labels and [AAD Pod Identity](https://github.com/Azure/aad-pod-identity)
```
Comment (weng271190436) [inline README.md:25]: You don't have to use AAD Pod Identity though. If there is no AAD pod identity installed and you just assign the MSI (with AcrPull permission to the ACR) to your agent pool it should also work

File: charts/msi-acrpull/values.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+# armAuth:
+#   identityResourceID: "<you managed identity resource id>"
+#   identityClientID: "<you managed identity client id>"
+
+msiacrpull:
+  image:
+    repository: mcr.microsoft.com/aks/msi-acrpull
+    pullPolicy: IfNotPresent
+    # Overrides the image tag whose default is the chart appVersion.
+    tag: v0.1.0-alpha
```
Comment (weng271190436) [inline charts/msi-acrpull/values.yaml:17]: not your problem but we have v0.1.0-beta now. I think I even released later version internally but haven't released it on github and now I lost permission.

**Review by weng271190436 — COMMENTED** [review]
I actually don't have permission to merge this anymore. @paulgmiller wondering who should merge this now

File: charts/msi-acrpull/crds/crd.yaml (hunk 1 above)
Comment (dmakeroam) [reply charts/msi-acrpull/crds/crd.yaml:6]: OK, I will remove the annotation to have helm support only on helm3

File: README.md (hunk 2 above)
Comment (dmakeroam) [reply README.md:25]: Oh, I never try this way. I assigned the MSI to user pool and it doesn't work. That is why AAD Pod Identity came to my solution.

Comment (dmakeroam) [reply README.md:25]: Let me check

**Comment by paulgmiller** [comment]
> I actually don't have permission to merge this anymore. @paulgmiller wondering who should merge this now

I think @samkreter @gambtho  and isreal are the only admins. Seeing if I can join aks-overlay group.

**Comment by dmakeroam** [comment]
Any updates on this? @paulgmiller

**Comment by stevekuznetsov** [comment]
Superseded by #80

**Pull request closed without merging**